File: LeetCode/349/solution.cpp
// 349. Intersection of Two Arrays
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int> mp;
        vector<int> ans;

        for(auto a: nums1) mp[a]++;
        for(auto a: nums2) {
            if(mp[a]) {
                ans.push_back(a);
                mp[a] = 0;
            }
        }

        return ans;
    }
};

File: LeetCode/349/solution.java
// 349. Intersection of Two Arrays
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Map<Integer, Integer> map = new HashMap<>();
        List<Integer> ans = new ArrayList<>();

        for(int a: nums1) map.put(a, 1);
        for(int a: nums2) {
            if(map.containsKey(a) && map.get(a)==1) {
                ans.add(a);
                map.put(a, 0);
            }
        }

        return ans.stream().mapToInt(i->i).toArray();
    }
}